src/EventGridExtension/EventGridExtensionConfig.cs

```diff
@@ -17,25 +19,30 @@ namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
     public class EventGridExtensionConfig : IExtensionConfigProvider,
                        IAsyncConverter<HttpRequestMessage, HttpResponseMessage>
     {
-        private TraceWriter _tracer = null;
+        private ILogger _logger;
 
         public void Initialize(ExtensionConfigContext context)
         {
             if (context == null)
             {
                 throw new ArgumentNullException("context");
             }
-            else if (context.Trace == null)
-            {
-                throw new ArgumentNullException("context.Trace");
-            }
-            _tracer = context.Trace;
+
+            _logger = context.Config.LoggerFactory.CreateLogger<EventGridExtensionConfig>();
 
             Uri url = context.GetWebhookHandler();
-            _tracer.Trace(new TraceEvent(System.Diagnostics.TraceLevel.Info, $"registered EventGrid Endpoint = {url}"));
+            _logger.LogInformation($"registered EventGrid Endpoint = {url?.GetLeftPart(UriPartial.Path)}");
+
+            // use converterManager as a hashTable
+            // also take benefit of identity converter
+            context
+                .AddBindingRule<EventGridTriggerAttribute>() // following converters are for EventGridTriggerAttribute only
+                .AddConverter<JObject, string>((jobject) => jobject.ToString(Formatting.Indented))
+                .AddConverter<string, JObject>((str) => JObject.Parse(str)) // used for direct invocation
+                .AddConverter<JObject, EventGridEvent>((jobject) => jobject.ToObject<EventGridEvent>()) // surface the type to function runtime
```

Comment:
Is the EventGridEvent rule covered by the POCO rule?

Reply:
it is, but I think if I want to use "EventGridEvent" in csharp script with `#r`,  I need to declare a specific converter?

![image](https://user-images.githubusercontent.com/6531400/36824936-be1f54f4-1cb9-11e8-8957-cd45228a87a0.png)